pkg/apis/testing/v1alpha1/testsuite_types.go

```diff
@@ -0,0 +1,138 @@
+	ComponentsSelector []string `json:"componentsSelector,omitempty"`
+	// Running all tests from suite cannot take more time that specified here
+	SuiteTimeout *metav1.Duration `json:"suiteTimeout,omitempty"`
+	// If specific TestDefinition does not define timeout, use this one
+	DefaultTestTimeout *metav1.Duration `json:"defaultTestTimeout,omitempty"`
+	// Should I repeat every test? Default value will be 1
+	Repeat int64 `json:"repeat,omitempty"`
+	// In case of a failed test, how many times it will be retried.
+	// If test failed and on retry it succeeded, Test Suite should be marked as a succeeded.
+	MaxRetries int64 `json:"maxRetries,omitempty"`
+}
+
+type NamespacedTest struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+// TestSuiteStatus defines the observed state of TestSuite
+type TestSuiteStatus struct {
+	StartTime      *metav1.Time         `json:"startTime,omitempty"`
+	CompletionTime *metav1.Time         `json:"completionTime,omitempty"`
+	Conditions     []TestSuiteCondition `json:"conditions,omitempty"`
+	Results        []TestResult         `json:"results,omitempty"`
+}
+
+type TestSuiteConditionType string
+type Status string
+type TestResultConditionType string
+
+const (
+	StatusTrue    Status = "true"
+	StatusFalse   Status = "false"
+	StatusUnknown Status = "unknown"
+
+	// SuiteUninitialized is when suite has not yet determined tests to run
+	SuiteUninitialized TestSuiteConditionType = "pending"
+	// When test are running
+	SuiteRunning TestSuiteConditionType = "running"
+	// When suite is finished and there were configuration problems, like missing test image
+	SuiteError TestSuiteConditionType = "error"
+	// When suite is finished and there were failing tests
+	SuiteFailed TestSuiteConditionType = "failed"
+	// When all tests passed
+	SuiteSucceed TestSuiteConditionType = "succeed"
+
+	// Test is not yet scheduled
+	TestPending TestResultConditionType = "pending"
+	// Test is running
+	TestRunning TestResultConditionType = "running"
+	TestError   TestResultConditionType = "error"
+	TestFailed  TestResultConditionType = "failed"
+	TestSucceed TestResultConditionType = "succeed"
+)
+
+type TestSuiteCondition struct {
+	Type    TestSuiteConditionType `json:"type"`
+	Status  Status                 `json:"status"`
+	Reason  string                 `json:"reason"`
+	Message string                 `json:"message"`
+}
+
+type TestResultCondition struct {
+	Type    TestResultConditionType `json:"type"`
+	Status  Status                  `json:"status"`
+	Reason  string                  `json:"reason"`
+	Message string                  `json:"message"`
+}
+
+type TestResult struct {
+	// Test name
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+	// Unique ID of specific test execution. Equivalent to testing Pod name
+	ID         string                `json:"id"`
+	Conditions []TestResultCondition `json:"conditions,omitempty"`
```

Comment:
IMO having conditions array here is a little bit too much:

```json

{
  "status": {
    "conditions": [
      {
        "type": "abc",
        "status": "abc",
        "reason": "abc",
        "message": "abc"
      },
      {
        "type": "abc1",
        "status": "abc",
        "reason": "abc",
        "message": "abc"
      }
    ],
    "results": [
      {
        "name": "n",
        "namespace": "abc",
        "ID": "abc",
        "conditions": [
          {
            "type": "abc",
            "status": "abc",
            "reason": "abc",
            "message": "abc"
          },
          {
            "type": "abc1",
            "status": "abc",
            "reason": "abc",
            "message": "abc"
          }
        ]
      }
    ]
  }
}
```

IMO in TestResult the final state should be inlined

Reply:
What is the difference between TestSuiteCondition and TestResultCondition

Reply:
As @sjanota said, I was trying to be consistent